### PR TITLE
Travis improvements.

### PR DESCRIPTION
**Pull request by adelton**
* Check that we only change Dockerfiles that we test with Travis CI.
* Run separate build jobs for each Dockerfile.
* If only tracked Dockerfile is modified, only run the `docker build` for that Dockerfile.

### CODE REVIEW

**Comment by stlaz** [comment]
Nicely done!

**Comment by adelton** [comment]
https://travis-ci.org/freeipa/freeipa-container/builds/400091320 shows the new output format.

**Comment by adelton** [comment]
And yes, changing `Dockerfile.rhel-7` will cause it to complain that we modify Dockerfile that Travis does not cover ... not sure how much we care about that ...

File: .travis.yml [hunk 1]
```diff
@@ -5,10 +5,35 @@ sudo: required
...
-- for i in $DOCKERFILES_CHANGED ; do docker build -f $i . ; done
+
+stages:
+- sanity
+- build
+
+matrix:
+  include:
+    - stage: sanity
+      script: if comm -23 <( echo "$FILES_CHANGED" ) <( echo "$TRAVIS_DOCKERFILES" ) | grep Dockerfile. ; then echo "Dockerfile modified but not tracked by Travis." >&2 ; exit 1 ; else echo "No unexpected Dockerfile changes, OK." ; fi
```
Comment (stlaz) [inline .travis.yml]: Could you use some kind of multi-line string to make this a bit more readable?
https://stackoverflow.com/a/21699210

Comment (adelton) [reply .travis.yml]: I could but what is not very readable then is Travis' output -- the formatting in the build log is then off.

**Comment by stlaz** [comment]
I doubt there's an easy way to test RHEL 7 image building in Travis, and usually these changes are those done to "downstream" first anyway, so I'd say we don't have to care about that that much.

**Comment by adelton** [comment]
We could do something like
```
if comm -23 <( echo "$FILES_CHANGED" | grep -v rhel ) <( echo "$TRAVIS_DOCKERFILES" ) | grep Dockerfile.
```
to ignore changes to `Dockerfile.rhel-7` ... but it might actually be nicer to have such pull requests fail and be reminded that the change was not tested by CI.

**Comment by stlaz** [comment]
I suppose

**Pull request merged**